File: AA/tetrahedron.py
n = int(raw_input())

mod = 1000000007

def calc(y):
	x = 1
	prev = 0
	threes = 1
	if y == 1:
		return 0
	elif y == 2:
		return 3
	
	for fraction in range(n-1):
		x = threes * 3
		threes *= 3
		threes = threes % mod
		x = x % mod
		x = x - prev
		prev = x
	
	return x % mod

print calc(n)

File: AA/cpp/tetrahedron.cpp
#include <iostream>
using namespace std;

  int mod = 1000000007ll;

  int main() {
       int N;
       cin >> N;

       long long ans = 0;
       if (N == 1) {
            cout << 0;
            return 0;
       }

       long long threes = 1;
       long long prev = 0;
       long long cur = 0;
       for (int i = 2; i <= N; i++) {
            threes = 3ll*threes;
            cur = threes - prev;
            if (cur < 0) cur +=mod;
            threes %= mod;
            cur %= mod;
            prev = cur;
       }

       cout << cur;
       return 0;
}